### DISCUSSION TITLE
Question: How to change DateTime to a spicific date

### DISCUSSION THREAD

**Discussion by krottv** [korlibs/korge#809] [Q&A]
Hi. Currently in Android I use the following code:

```
val calendarNew = Calendar.getInstance()

calendarNew.set(Calendar.DAY_OF_WEEK, 2)
calendarNew.set(Calendar.HOUR_OF_DAY, 18)
calendarNew.set(Calendar.MINUTE, 0)
calendarNew.roll(Calendar.WEEK_OF_YEAR, 1)

return calendarNew.timeInMillis
```

How can I do the same with KLock?

The purpose of this code is to take get time of the next week day when I need to send a notification.

**Comment by soywiz** [comment]
So you want to get a date for a specific day of the week (Monday, Tuesday...), for a specific week in the year?

And if that's the case:
Let's consider this year 2021. The first week started on Friday, and the second week on Monday. So requesting the Tuesday for the first week would be a date in 2020 or an error, right? And the second Tuesday would be 5th of January of 2021.

Is that correct?

**Comment by krottv** [comment]
@soywiz Not for a specific week of they year, just for the next. The idea is to set notify user the next week at convenient time. For example. Current datetime - 09:08 am, 25th of July 2021. Then the date returned by this function would be:
06:00 pm, 27th of July 2021. (because we rolled to the next week 1st august, which is sunday. And we pick specific day of this week, which is 27th)

**Comment by soywiz** [comment]
In next version of Klock, you will be able to do this:

```
(Date(2021, Month.July, 25) + 1.weeks).inThisWeek(DayOfWeek.Monday, KlockLocale.yourLocale)
```

It is locale-dependant because in some places, week starts on Sunday, in others in Monday. Hope it helps!

**Comment by krottv** [comment]
@soywiz That would be awesome! Thanks

**Comment by krottv** [comment]
@soywiz 
Also there should be a way to change hour of day. Like:
(DateTime.now() + 1.weeks).inThisWeek(DayOfWeek.Monday, KlockLocale.yourLocale).inThisHourOfDay(18, KlockLocale.yourLocale)

**Comment by soywiz** [comment]
The function I created works with `Date`, not with `DateTime`. You can construct your DateTime from the Date part, and specifying the time that is right for you

**Comment by krottv** [comment]
@soywiz yes, that would work
